axelor-account/src/main/java/com/axelor/apps/account/web/AccountingReportController.java

```diff
@@ -238,4 +238,39 @@ public void showMoveExported(ActionRequest request, ActionResponse response) {
 
     response.setView(actionViewBuilder.map());
   }
+
+  /**
+   * @param request
+   * @param response
+   */
+  public void generateDocumentation(ActionRequest request, ActionResponse response) {
+    try {
+      AccountingReport accountingReport = request.getContext().asType(AccountingReport.class);
+      accountingReport = accountingReportRepo.find(accountingReport.getId());
+
+      if (accountingReport.getExportTypeSelect() == null
+          || accountingReport.getExportTypeSelect().isEmpty()
+          || accountingReport.getTypeSelect() == 0) {
+        response.setFlash(I18n.get(IExceptionMessage.ACCOUNTING_REPORT_4));
+        response.setReload(true);
+        return;
+      }
+      accountingReportService.setPublicationDateTime(accountingReport);
+
+      String name = "FEC Documentation " + accountingReport.getRef();
+
+      String fileLink =
+          ReportFactory.createReport("DescriptiveDocumentationFEC.rptdesign", name + "-${date}")
+              .addParam("AccountingReportId", accountingReport.getId())
+              .addParam("Locale", ReportSettings.getPrintingLocale(null))
+              .addFormat("odt")
+              .toAttach(accountingReport)
+              .generate()
+              .getFileLink();
```

Comment:
This code belongs in a service. I think the date might not be saved because there is no active transaction.